src/http/HttpRequest.ts

```diff
@@ -39,15 +50,51 @@ export class HttpRequest implements types.HttpRequest {
                 body = init.body.string;
             }
 
-            this.#uReq = new uRequest(url, {
+            uReq = new uRequest(url, {
                 body,
                 method: nonNullProp(init, 'method'),
                 headers: fromNullableMapping(init.nullableHeaders, init.headers),
             });
         }
 
-        this.query = new URLSearchParams(fromNullableMapping(init.nullableQuery, init.query));
-        this.params = fromNullableMapping(init.nullableParams, init.params);
+        const query = new URLSearchParams(fromNullableMapping(init.nullableQuery, init.query));
+        const params = fromNullableMapping(init.nullableParams, init.params);
+
+        return [uReq, query, params];
+    }
+
+    #initStreamRequest(init: InternalHttpRequestInit): [uRequest, URLSearchParams, HttpRequestParams] {
+        const proxyReq = nonNullProp(init, 'proxyRequest');
+
+        const hostHeaderName = 'x-forwarded-host';

```

Comment:
I would encapsulate lines 69-75 in a private function (`constructURL()`?) and then line 76 could just be `const url = constructURL(proxyReq);`.

Reply:
I would normally encapsulate something when either (a) it needs to be used in two places or (b) the method is getting too long. This code is only used once and I guess I feel like 20-30 lines is perfectly fine

Otherwise I find it more readable to just have one method and use spacing/comments to delineate sections if necessary. It's personal preference, but I just don't like navigating a whole call stack of methods if I want to see the full code for a (reasonably sized) method. Those methods could get out of order in the file, have other methods with confusingly similar names, and make it more difficult to refactor in the future.